### PR TITLE
fix(SPV-000): BUMP verification changes

### PR DESCRIPTION
**Pull request by jakubmkowalski**
This pull request includes several changes to the `beef/bump.go` and `spv/spv_test.go` files to improve the calculation of the Merkle root and refactor the code for better error handling and readability. The most important changes include adding a new method for calculating from children, modifying the loop to handle previous levels, and reordering imports.

Improvements to Merkle root calculation:

* [`beef/bump.go`](diffhunk://#diff-f48b7c5d8430d414ae2da75f347a3a659d85621037ac8d638e244ef81258a21aR113-R134): Added the `calculateFromChildren` method to compute a leaf from its children when it is not found in the current level.
* [`beef/bump.go`](diffhunk://#diff-f48b7c5d8430d414ae2da75f347a3a659d85621037ac8d638e244ef81258a21aL59-L75): Modified the loop in `calculateMerkleRoot` to handle previous levels and compute missing leaves using the new method.

# Pull Request Checklist

- [x] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/bitcoin-sv/go-paymail/blob/main/.github/CODE_STANDARDS.md)
- [x] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/bitcoin-sv/go-paymail/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] 🏠 I tested my changes locally.
- [x] ✅ I have provided tests for my changes.
- [x] 📝 I have used conventional commits.
- [ ] 📗 I have updated any related documentation.
- [ ] 💾 PR was issued based on the Github or Jira issue.

### CODE REVIEW

File: beef/bump.go [hunk 1]
```diff
@@ -56,23 +57,34 @@ func calculateMerkleRoot(baseLeaf BUMPLeaf, bump BUMP) (string, error) {
...
+			var err error
+			if previousBLevel != nil {
+				leafInPair, err = calculateFromChildren(newOffset, previousBLevel)
+				if err != nil {
+					return "", err
+				}
+			} else {
+				return "", errors.New("cannot compute leaf from children at base level")
+			}
 		}
```
Comment (mgosek-4chain) [inline beef/bump.go:79]: I made a small adjustment to the code structure, and after running spv_test, everything seems to be working as expected 😄. 

```
if leafInPair == nil {
	if previousBLevel == nil {
		return "", errors.New("cannot compute leaf from children at base level")
	}
			
	var err error
	leafInPair, err = calculateFromChildren(newOffset, previousBLevel)
	if err != nil {
		return "", err
	}
}
```

File: beef/bump.go [hunk 2]
```diff
@@ -98,6 +110,28 @@ func findLeafByOffset(offset uint64, bumpLeaves []BUMPLeaf) *BUMPLeaf {
...
+	str, err := merkleTreeParentStr(leftNode, rightNode)
+	if err != nil {
+		return nil, errors.New("could not find pair")
+	}
+	return &BUMPLeaf{
+		Hash:   str,
+		Offset: offset,
+	}, nil
+}
+
```
Comment (mgosek-4chain) [inline beef/bump.go:134]: Do you think it would be a good idea to extract these errors into external variables and add some test cases to cover these edge cases? 🤔😄 Additionally, small code style suggestion to use new lines after statements like `if' etc.

**Review by wregulski — APPROVED** [review]
Nice findings @jakubmkowalski - good job!

File: beef/bump.go (hunk 1 above)
Comment (jakubmkowalski) [reply beef/bump.go]: fixed

File: beef/bump.go (hunk 2 above)
Comment (jakubmkowalski) [reply beef/bump.go]: New lines after 'if's introduced. 
In near future we will be swapping functions regarding BUMP calculation to go-SDK ones.
Because this is a temporary fix, and such situations can occur in theory and I'm not sure if they can happen in a real life scenario, and it can be time-consuming to find the hexes for which we get such errors. 
Therefore, I would hold off on developing any test cases to get them.

**Review by mgosek-4chain — APPROVED** [review]
